Move compute table functions from FEComponentTransferSoftwareApplier to FEComponentTransfer

carlosgcampos opened the pull request:
#### 82f851ba6e367121f35f2ca6d2bdc96e944bbf83
<pre>
Move compute table functions from FEComponentTransferSoftwareApplier to FEComponentTransfer
<a href="https://bugs.webkit.org/show_bug.cgi?id=273817">https://bugs.webkit.org/show_bug.cgi?id=273817</a>

Reviewed by Nikolas Zimmermann.

Skia implementation will need those functions too, by moving them we
will be able to use them without duplicating the code.

* Source/WebCore/platform/graphics/filters/FEComponentTransfer.cpp:
(WebCore::FEComponentTransfer::computeLookupTable):
* Source/WebCore/platform/graphics/filters/FEComponentTransfer.h:
* Source/WebCore/platform/graphics/filters/software/FEComponentTransferSoftwareApplier.cpp:
(WebCore::FEComponentTransferSoftwareApplier::applyPlatform const):
(WebCore::FEComponentTransferSoftwareApplier::computeIdentityTable): Deleted.
(WebCore::FEComponentTransferSoftwareApplier::computeTabularTable): Deleted.
(WebCore::FEComponentTransferSoftwareApplier::computeDiscreteTable): Deleted.
(WebCore::FEComponentTransferSoftwareApplier::computeLinearTable): Deleted.
(WebCore::FEComponentTransferSoftwareApplier::computeGammaTable): Deleted.
(WebCore::FEComponentTransferSoftwareApplier::computeLookupTable): Deleted.
* Source/WebCore/platform/graphics/filters/software/FEComponentTransferSoftwareApplier.h:

Canonical link: <a href="https://commits.webkit.org/278458@main">https://commits.webkit.org/278458@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9fbf3cfc5060e36119419cc5f5914a77249b8743

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/50605 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/29902 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/2919 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/53864 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/1295 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/52908 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/36158 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/946 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/41254 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/52704 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/27550 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/43591 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22365 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/24940 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/840 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/9045 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/46934 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/903 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/55453 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/25706 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Debug-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/48664 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/26964 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/43729 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/47720 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11088 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/27830 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/26696 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

GeorgesStavracas reviewed (commented):
Looks good, will definitely shrink some code from #28176

nikolaszimmermann reviewed (approved):
r=me.

carlosgcampos committed:
Move compute table functions from FEComponentTransferSoftwareApplier to FEComponentTransfer
https://bugs.webkit.org/show_bug.cgi?id=273817

Reviewed by Nikolas Zimmermann.

Skia implementation will need those functions too, by moving them we
will be able to use them without duplicating the code.

* Source/WebCore/platform/graphics/filters/FEComponentTransfer.cpp:
(WebCore::FEComponentTransfer::computeLookupTable):
* Source/WebCore/platform/graphics/filters/FEComponentTransfer.h:
* Source/WebCore/platform/graphics/filters/software/FEComponentTransferSoftwareApplier.cpp:
(WebCore::FEComponentTransferSoftwareApplier::applyPlatform const):
(WebCore::FEComponentTransferSoftwareApplier::computeIdentityTable): Deleted.
(WebCore::FEComponentTransferSoftwareApplier::computeTabularTable): Deleted.
(WebCore::FEComponentTransferSoftwareApplier::computeDiscreteTable): Deleted.
(WebCore::FEComponentTransferSoftwareApplier::computeLinearTable): Deleted.
(WebCore::FEComponentTransferSoftwareApplier::computeGammaTable): Deleted.
(WebCore::FEComponentTransferSoftwareApplier::computeLookupTable): Deleted.
* Source/WebCore/platform/graphics/filters/software/FEComponentTransferSoftwareApplier.h:

Canonical link: https://commits.webkit.org/278458@main
